src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -1,51 +1,181 @@
 package io.pinecone.integration.controlPlane.pod;
 
 import io.pinecone.clients.Pinecone;
+import io.pinecone.exceptions.PineconeBadRequestException;
+import io.pinecone.exceptions.PineconeUnmappedHttpException;
 import io.pinecone.helpers.RandomStringBuilder;
-import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.openapitools.client.model.*;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
+import java.util.Random;
 
-public class CreateDescribeListAndDeleteIndexTest {
-    private Pinecone controlPlaneClient;
-    private final String apiKey = System.getenv("PINECONE_API_KEY");
-    private final String environment = System.getenv("PINECONE_ENVIRONMENT");
+import static io.pinecone.helpers.IndexManager.waitUntilIndexIsReady;
+import static org.junit.jupiter.api.Assertions.*;
 
-    @BeforeEach
-    public void setUp() {
-        controlPlaneClient = new Pinecone(apiKey);
-    }
+public class CreateDescribeListAndDeleteIndexTest {
+    private static final String environment = System.getenv("PINECONE_ENVIRONMENT");
+    private static final String indexName = RandomStringBuilder.build("create-index", 8);
+    private static Pinecone controlPlaneClient = new Pinecone(System.getenv("PINECONE_API_KEY"));
 
-    @Test
-    public void createAndDelete() throws InterruptedException {
-        String indexName = RandomStringBuilder.build("index-name", 8);
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        // Create the index
         CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
```

Comment:
Total nit, but would it be more idiomatic to have a constant that holds the `podType` instead of hardcoding it here & then again in the `describeAndListIndex` test below? (Not sure if Java has a concept of constants, tbf)

Reply:
I can set up some `static final` variables, I think that's primarily how Java deals with the idea of constants.